src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/TdsParserStateObjectManaged.cs

```diff
@@ -88,15 +89,26 @@ protected override uint SNIPacketGetData(PacketHandle packet, byte[] inBuff, ref
             string cachedFQDN,
             ref SQLDNSInfo pendingDNSInfo,
             string serverSPN,
+            DataSource details,
             bool isIntegratedSecurity,
             bool tlsFirst,
             string hostNameInCertificate,
             string serverCertificateFilename)
         {
-            SNIHandle? sessionHandle = SNIProxy.CreateConnectionHandle(serverName, ignoreSniOpenTimeout, timerExpire, out instanceName, ref spnBuffer, serverSPN,
-                flushCache, async, parallel, isIntegratedSecurity, iPAddressPreference, cachedFQDN, ref pendingDNSInfo, tlsFirst,
-                hostNameInCertificate, serverCertificateFilename);
+            SNIHandle? sessionHandle;
+            instanceName = new byte[1];
 
+                if (details == null)
+                {
+                    sessionHandle = null;
```

Comment:
Or maybe the block can be removed...

Reply:
Code will enter SNIProxy.CreateConnectionHandle and throw SNI errors. details are not expected to be null, previous code needed the check because SNILoadHandle.SingletonInstance was keeping track of LastError and details could be null, now exception is thrown.